Finished Mailroom2, dict lab, and file exercise

gjkim44 opened the pull request:
I am still working on trigrams

gjkim44 committed:
changed to show donation amounts and corrected float issues by changing int to float

gjkim44 committed:
just editing mailroom

gjkim44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

gjkim44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

gjkim44 committed:
finished dictionary lab

gjkim44 committed:
adding sherlock text

gjkim44 committed:
adding trigram from class

gjkim44 committed:
glitched happend in vscode having to re add files

gjkim44 committed:
tested mailroom and produced files for donors.  Still working on file_exercise and trigrams

gjkim44 committed:
updated files , amde changes using dictionary keys for mailroom, working on files exercise

gjkim44 committed:
deleting text files

gjkim44 committed:
finished file exercise added jpg file to test copying file and students.txt

gjkim44 committed:
changed formating to display data

gjkim44 committed:
updated file exercise to show specific list

gjkim44 committed:
added logic to account for 'nothing' language'

gjkim44 committed:
cleaned up file exercise and added punctuation removal for trigrams

gjkim44 committed:
text file to test punc removal

gjkim44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

gjkim44 committed:
finished trigrams with Chris's solution to finish assingment

nairraghav commented on students/GKim/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,110 @@
+    dct.pop("cake")
+    print("\nDictionary with cake removed")
+    show_dict(dct)
+    dct.update(fruit = "Mango")
```
you can also do dct['fruit'] = 'Mango'

nairraghav commented on students/GKim/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,110 @@
+    print("\n{x} {y} {x}".format( x = "*"*10, y = "END"))
+       
+    if "cake" in dct:
+        print("GOT ME SOME!")
```
🍰

nairraghav commented on students/GKim/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,110 @@
+    
+def dictionaries_2(dct):
+    
+    for key in dct.keys():
```
You actually don't need to specify the .keys(), you can just do
for key in dct:

nairraghav commented on students/GKim/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,110 @@
+    
+    for key in dct.keys():
+        t_count = 0
+        for val in dct[key]:
```
you can use the .count function instead of a forloop here 
ex: dct[key].lower().count('t')

nairraghav commented on students/GKim/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,110 @@
+    sets_2()
+
+if __name__ == "__main__": main()
+    
```
👏 👏  nice job!

nairraghav commented on students/GKim/lesson04/mailroom2.py:
```diff
@@ -0,0 +1,208 @@
+        else:
+            idx = len(main_donors)-1
+            in_main_donors = False
+            for x in range(len(main_donors)-1, -1, -1):
```
Since we are working with a dict now, you don't need to use a for loop to iterate through it. You can just check if the key exists (where the key would be the name)

nairraghav commented on students/GKim/lesson04/mailroom2.py:
```diff
@@ -0,0 +1,208 @@
+                    in_main_donors = True
+                    print("\nThe Donor you have selected is {}".format(thanks_answer))
+                    donation_amount = float(input("\nPlease enter the amount that {} kindly donated: ".format(thanks_answer)))
+                    main_donors[idx]["donations"].append(donation_amount)
```
This throws an error, try to run this locally and see what happens when you add a new donor multiple times.

nairraghav commented on students/GKim/lesson04/mailroom2.py:
```diff
@@ -0,0 +1,208 @@
+
+    return donor_stats
+
+def create_report():
```
After generating a new donor, this throws an error.

nairraghav commented on students/GKim/lesson04/trigram.py:
```diff
@@ -0,0 +1,96 @@
+
+import random
+
+punctuations = "!" + "-" + "," + "." + "(" + ")" + '"' + '?' + ';' + ":" + '\n'
```
import string

string.punctuation # this string contains all punctuation

nairraghav commented:
Overall, your dict lab and trigrams look good! Please review your mailroom pt 2, I was having some trouble running it as there were errors involved when adding new donors. I'll merge this in for now and have given credit for dict/trigrams

gjkim44 commented:
Hey Ron,
I am running the code on my end and it works, what to do?

On Tue, May 7, 2019 at 8:09 PM nairraghav <notifications@github.com> wrote:

> *@nairraghav* commented on this pull request.
> ------------------------------
>
> In students/GKim/lesson04/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/108#discussion_r281899459>
> :
>
> > @@ -0,0 +1,96 @@
> +
> +# WORDS = "with his head sunk upon his chest and his hands clasped behind him.".split()
> +# WORDS = "I wish I may I wish I might".split()
> +
> +import random
> +
> +punctuations = "!" + "-" + "," + "." + "(" + ")" + '"' + '?' + ';' + ":" + '\n'
>
> import string
>
> string.punctuation # this string contains all punctuation
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/108#pullrequestreview-234847881>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKAHIV4ZNKQLPE3HCPDVS5LPUI77LANCNFSM4HK5FYKQ>
> .
>


-- 
George J. Kim
